src/plugins/intel_cpu/src/memory_control.hpp

```diff
@@ -42,21 +42,21 @@ class MemoryControl {
     void releaseMemory();
 
 private:
-    explicit MemoryControl(std::vector<size_t> syncInds);
-    void insert(const MemoryRegion& region);
+    explicit MemoryControl();
+    void insert(const MemoryRegion& region, const std::vector<size_t>& syncInds);
 
     friend class NetworkMemoryControl;
 
 private:
-    std::vector<size_t> m_syncInds;
     std::vector<RegionHandlerPtr> m_handlers;
     bool m_allocated = false;
 };
 
 class NetworkMemoryControl {
 public:
     NetworkMemoryControl() = default;
-    MemoryControl& createMemoryControlUnit(std::vector<size_t> syncInds);
+    // @todo return std::reference_wrapper instead?
+    MemoryControl* createMemoryControlUnit();
```

Comment:
Why can't we simply return a reference?

Reply:
We can
Just do not like the idea of storing a plain reference inside the GraphContext